torchrl/envs/transforms/transforms.py

```diff
@@ -2433,3 +2433,121 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(decay={self.decay:4.4f},"
             f"eps={self.eps:4.4f}, keys={self.in_keys})"
         )
+
+
+class RewardSum(Transform):
+    """Tracks the accumulated rewards of each episode.
+
+    This transform accepts a list of tensordict keys ´in_keys´, and tracks their cumulative value along each
+    episode. The transform creates a new tensordict key for each input_key called ´episode_{input_key}´ where
+    the cumulative values are written.
+
+    If no in_keys are specified, this transform assumes ´reward´ to be the input key. However, multiple rewards
+    (e.g. reward1 and reward2) can also be specified in the. If ´in_keys´ are not present in the provided
+    tensordict, this transform hos no effect.
+    """
+
+    inplace = True
+
+    def __init__(
+        self,
+        in_keys: Optional[Sequence[str]] = None,
+        out_keys: Optional[Sequence[str]] = None,
+    ):
+        """Initialises the transform. Filters out non-reward input keys and defines output keys."""
+        if in_keys is None:
+            in_keys = ["reward"]
+        out_keys = [f"episode_{in_key}" for in_key in in_keys]
+
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Resets episode rewards."""
+        # Non-batched environments
+        if len(tensordict.batch_size) < 1 or tensordict.batch_size[0] == 1:
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key] = 0.0
+
+        # Batched environments
+        elif "reset_workers" in tensordict.keys():
```

Comment:
I guess that the default in the batched cased where no "reset_workers" is present would be to reset them all no?
otherwise in multi-agent settings for instance, a user could do

```
env.reset()
```
an end up with an episode reward that keeps on piling up

Reply:
you are right. Now if no "reset_workers" is found in the tensordict for the batched case, all envs are reset